main.go

```diff
@@ -179,6 +180,29 @@ func main() {
 		os.Exit(1)
 	}
 
+	decoder := admission.NewDecoder(mgr.GetScheme())
+
+	if os.Getenv("DISABLE_AUTO_ANNOTATION") != "true" {
```

Comment:
nit: If we define these env variables as global vars, it would make it easier in the long run to update them and re-use them. `"DISABLE_AUTO_ANNOTATION_KEY = "DISABLE_AUTO_ANNOTATION"`